cmd/talosctl/cmd/mgmt/cluster/create.go

```diff
@@ -127,6 +158,14 @@ var createCmd = &cobra.Command{
 	Short: "Creates a local docker-based or QEMU-based kubernetes cluster",
 	Long:  ``,
 	Args:  cobra.NoArgs,
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		definedGenFlag := checkForDefinedGenFlag(cmd.Flags())
+		if definedGenFlag != "" {
```

Comment:
we could probably move this check closer to the code which checks for `input-dir` flag presence vs. config generqtion

Reply:
I tried to do the check where you suggest, but there was a problem. That check requires access to the flags. If I use createCmd.Flags() inside the create() function I get an "initialization cycle" because create() is used inside the initialization of createCmd.
create() only accepts a context.Context as input, so I considered passing in the flags there, but that would mean using a type assertion inside create() when I get the flags out again from the context.Context.
That's why I decided to do it in the PreRunE.

Now I changed the signature of create, since it's only used exactly once and the context.Background() happens inside create.